### ISSUE TITLE
Object reference not set to an instance of an object. when trying to save module settings

### ISSUE THREAD

**Issue by micahsdad1402** [nbrightproject/NBrightBuy#54]
This seems to be happening with some NBS modules. I tested them all and these failed.
NBS_CategoryMenu, NBS_MiniCart, NBS_ProfileForm, NBS_DisplaySearch

However, it's all my my new live site, not on my development one.

Any ideas?

`at DotNetNuke.Security.Permissions.PermissionController.<>c__DisplayClass8_0.b__0(PermissionInfo p)
   at System.Linq.Enumerable.WhereListIterator`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at DotNetNuke.Security.Permissions.PermissionController.GetPermissionsByModule(Int32 moduleId, Int32 tabId)
   at DotNetNuke.Security.Permissions.Controls.ModulePermissionsGrid.GetPermissions()
   at DotNetNuke.Security.Permissions.Controls.PermissionsGrid.CreateChildControls()
   at DotNetNuke.Security.Permissions.Controls.ModulePermissionsGrid.CreateChildControls()
   at System.Web.UI.Control.EnsureChildControls()
   at System.Web.UI.Control.FindControl(String id, Int32 pathOffset)
   at System.Web.UI.Control.FindControl(String id, Int32 pathOffset)
   at System.Web.UI.Control.FindControl(String id, Int32 pathOffset)
   at System.Web.UI.Control.FindControl(String id, Int32 pathOffset)
   at System.Web.UI.Control.FindControl(String id, Int32 pathOffset)
   at System.Web.UI.Page.FindControl(String id)
   at System.Web.UI.Page.ProcessPostData(NameValueCollection postData, Boolean fBeforeLoad)
   at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)`

**Comment by micahsdad1402** [comment]
Dave,
Have you got any ideas on this?
I'm guessing it's some sort of DNN setting somewhere

**Comment by leedavi** [comment]
Err!! NO

I've never seen this.  Does it end up in the event log or does it show on screen?

**Comment by leedavi** [comment]
What edit permissions do the modules have?

**Comment by micahsdad1402** [comment]
I got around this by disabling the inherit permissions from the page and then just specified the same settings for the module.

**Comment by leedavi** [comment]
OK, strange I'm sure I usually just inherit.  I'll test that and see if I can recreate.

**Comment by micahsdad1402** [comment]
It's also only happening on my live server, not on my notebook where I'm developing, so it's probably data/setting related. 

It was also only happening on some of the modules, and all NBS ones, not that I have added many others at this stage. At this stage I have a work around, so it's not a big priority now. Was awkward before because I couldn't change settings. It also wasn't from the start, but later on that I started having this issue.
